ncm-spma/src/test/perl/apt-cleanup-sources.t

```diff
@@ -0,0 +1,117 @@
+# -*- mode: cperl -*-
+# ${license-info}
+# ${author-info}
+# ${build-info}
+
+=pod
+
+=head1 DESCRIPTION
+
+Tests for the cleanup_old_sources method.  This method ensures that
+outdated sources in /etc/apt/sources.list.d/ are removed.
+
+=head1 TESTS
+
+=cut
+
+use strict;
+use warnings;
+use Readonly;
+use English;
+
+use Test::Quattor;
+use Test::More;
+use Test::MockModule;
+use NCM::Component::spma::apt;
+use CAF::Path;
+use CAF::FileWriter;
+use Set::Scalar;
+
+
+Readonly my $SOURCE_DIR => "target/test/clean.sources.list.d";
+
+my $sources = [{
+    name => "source1",
+    owner => 'localuser@localdomain',
+    protocols => [{
+        name => "http",
+        url => "http://localhost.localdomain",
+    }],
+}];
+
+my $cmp = NCM::Component::spma::apt->new("spma");
+my $mock = Test::MockModule->new('NCM::Component::spma::apt');
+
+my $immutable;
+my $directory_exists;
+my $sourcefiles;
+
+
+$mock->mock('_glob', sub {
+    return @$sourcefiles;
+});
+
+
+$mock->mock('cleanup', sub {
+    shift;
+    if (!$immutable) {
+        my $filename = shift;
+        $sourcefiles->delete($filename) || return undef;
+        return 1;
+    };
+    return undef;
+});
+
+
+$mock->mock('directory_exists', sub {
+    return $directory_exists;
+});
+
+
+=pod
+
+=head2 The source directory doesn't exist
+
+If the source directory doesn't exist, the error should be
+reported.
+
+=cut
+
+$immutable = 0;
+$directory_exists = 0;
+$sourcefiles = Set::Scalar->new();
+
+is($cmp->cleanup_old_sources($SOURCE_DIR), 0, "Throw error if source dir does not-exist source dir");
+is($cmp->{ERROR}, 1, "Report Non-existing source dir to the user");
+
+=pod
+
+=head2 The source exists and contains files that should be removed
+
+In this case, any unwanted files should be removed
+
+=cut
+
+$immutable = 0;
```

Comment:
remove these, just create the files, parent dir will exist.

```perl
foreach my $fn (qw(source1 source2)) {
  set_file_contents("$SOURCE_DIR/$_.list", "")
};
```
this will create the parent dir and "touch" the files.